spec/github_helper_spec.rb

```diff
@@ -217,31 +217,77 @@ def get_milestone(milestone_name:)
       allow(Octokit::Client).to receive(:new).and_return(client)
     end
 
-    it 'has the correct dates to code freeze without submission' do
-      comment = "Code freeze: October 22, 2022\nApp Store submission: November 15, 2022\nRelease: October 25, 2022\n"
-      options = { due_on: '2022-10-22T12:00:00Z', description: comment }
+    it 'computes the correct dates for standard period' do
+      due_date = '2022-10-20T08:00:00Z'.to_time.utc
+      options = {
+        due_on: '2022-10-20T12:00:00Z',
+        description: "Code freeze: October 20, 2022\nApp Store submission: October 24, 2022\nRelease: October 27, 2022\n"
+      }
 
       expect(client).to receive(:create_milestone).with(test_repo, test_milestone_number, options)
-      create_milestone(need_submission: false, milestone_duration: 24, days_code_freeze: 3)
+      create_milestone(due_date: due_date, days_until_submission: 4, days_until_release: 7)
     end
 
-    it 'has the correct dates to code freeze with submission' do
-      comment = "Code freeze: October 22, 2022\nApp Store submission: October 22, 2022\nRelease: October 25, 2022\n"
-      options = { due_on: '2022-10-22T12:00:00Z', description: comment }
+    it 'computes the correct dates when the due date is on the verge of a DST day change' do
+      # The PST to PDT (DST) change is made on the second Sunday in March and finishes on the first Sunday in November
+      Time.zone = 'Pacific Time (US & Canada)'
+      due_date = '2022-06-18T23:00:00Z'.to_time
```

Comment:
> \# The PST to PDT (DST) change is made on the second Sunday in March and finishes on the first Sunday in November

Then that means that the date you used (`2022-06-18` is not even close to a DST change then, since it's neither close to March nor November… so this actually tests the opposite case?

Reply:
I wanted to check the PST to PDT details and landed on Wikipedia. Here's [the link](https://en.wikipedia.org/wiki/Pacific_Time_Zone#:~:text=Effective%20in%20the%20U.S.%20in,the%20first%20Sunday%20in%20November.) for future reference, just in case.

Reply:
I had problems testing this properly, to be honest. It confuses me that the fact that PST/PDT changes are made at 2 am, Also for some reason, the date to not change when I use the 14 march (the actual DST change date) as my date, even forcing the timezone to be at PST. So I just used any date inside the DST changes, given the date that was put was on UTC.

Reply:
Due to the Ci failure, I think that this is not the proper way to enforce the use of a specific timezone (although it passes here on my local test run). I'll try to build it differently, I was researching other ways to do this and found this one, which is wrapping the test on the timezone change block:

```ruby
Time.use_zone('Europe/London') do
  due_date = Time.zone.parse('2022-03-27 23:00:00')
  # ...
end
```

TBH, I'm not sure if this is the correct way to do it either, as there are a lot of different answers on how to do this (most of them on rails), on StackOverflow. If you folks have suggestions on how to deal with this, I would love to hear @AliSoftware @mokagio 😄 

Note: I'll use the DST from Europe this time, that makes more sense in my head than the American ones. 😅

Reply:
From the top of my head I don't remember what's the right way to do it, though `Time.use_zoen(…) do…end` feels like a good lead. In any case, I'd recommend picking a date that you know is before _both_ Europe and Pacific DST change dates, and use an offset for the date addition that is large enough to end up after _both_ DST change dates as well. E.g. a date at start of Oct and an offset of 60 days, or something like that, to be sure to cover both cases anyway, even if `Time.use_zone` should do the trick.